### PR TITLE
Reactnative/input validation

### PR DESCRIPTION
**Pull request by vivekvijayakrishnan**
- Completed https://github.com/Microsoft/AdaptiveCards/issues/1978
- Fixed couple of bugs

### CODE REVIEW

File: source/reactnative/src/adaptive-cards.js [hunk 1]
```diff
@@ -229,11 +234,24 @@ export default class AdaptiveCards extends React.Component {
 	}
 
 	// Invoke onExecuteAction if the consumer app provide it via props.
-	onExecuteAction = (action) => {
-		if (this.props.onExecuteAction) {
+	onExecuteAction = (action,showValidationErrors) => {
+		if(!showValidationErrors && !this.validateInputs())
+			this.setState({showErrors:true});
```
Comment (pragadeeshk) [inline source/reactnative/src/adaptive-cards.js]: setting showErrors as true when showValidationErrors is false looks contradictory. Can we give more appropriate variable name here?

File: source/reactnative/src/components/inputs/picker-input.js [hunk 2]
```diff
@@ -34,17 +36,32 @@ export class PickerInput extends React.Component {
 		this.type = Constants.EmptyString;
 		this.modalButtonText = Constants.DoneString;
 		this.parseHostConfig();
+		this.isValidationRequired=!!this.payload.validation &&
```
Comment (pragadeeshk) [inline source/reactnative/src/components/inputs/picker-input.js]: This assignment looks confusing. Add appropriate brackets

File: source/reactnative/src/adaptive-cards.js (hunk 1 above)
Comment (vivekvijayakrishnan) [reply source/reactnative/src/adaptive-cards.js]: sure will change it.

File: source/reactnative/src/utils/enums.js [hunk 3]
```diff
@@ -131,6 +131,12 @@ export const Height = Object.freeze({
 	Stretch: 1
 });
 
+export const ValidationNecessity= {
+    Optional: 'optional',
```
Comment (BalajiR) [inline source/reactnative/src/utils/enums.js]: Shall we use capitalized values for this `enum`.

File: source/reactnative/src/components/actions/action-button.js [hunk 4]
```diff
@@ -88,14 +88,20 @@ export class ActionButton extends React.Component {
      * @description Invoked for the action type Constants.ActionSubmit
      */
 	onSubmitActionCalled() {
-		let mergedObject = null;
-		if (this.data !== null) {
-			mergedObject = { ...this.data, ...this.context.inputArray };
-		} else {
-			mergedObject = { ...this.context.inputArray };
+		let mergedObject = {};
+		for (const key in this.context.inputArray) {
```
[the remark this answers is not in the document]
Comment (vivekvijayakrishnan) [reply source/reactnative/src/components/actions/action-button.js]: Yes. We need it. `inputArray` has the `errorstate` value. I don't want to include the same while we pass the action data to the consumer app.

File: source/reactnative/src/components/inputs/choice-set/choice-set-input.js [hunk 5]
```diff
@@ -39,20 +40,32 @@ export class ChoiceSetInput extends React.Component {
...
 		this.value = Constants.EmptyString;
 		this.choices = [];
-		this.payload = props.json
+		this.payload = props.json;
+		
+		this.isValidationRequired = !!this.payload.validation &&
+			(Enums.ValidationNecessity.Required == this.payload.validation.necessity ||
+				Enums.ValidationNecessity.RequiredWithVisualCue == this.payload.validation.necessity);
+
+		this.validationText = (this.payload.validation && this.payload.validation.validationFailedText) ?
```
[the remark this answers is not in the document]
Comment (vivekvijayakrishnan) [reply source/reactnative/src/components/inputs/choice-set/choice-set-input.js]: done

[the remark this answers is not in the document]
Comment (vivekvijayakrishnan) [reply source/reactnative/src/components/inputs/choice-set/choice-set-input.js]: This was in the recent discussion. I missed it, thanks for finding this. +1

File: source/reactnative/src/adaptive-cards.js [hunk 6]
```diff
@@ -229,11 +234,24 @@ export default class AdaptiveCards extends React.Component {
...
+		else if(this.props.onExecuteAction) {
 			this.props.onExecuteAction(action);
 		}
 	}
+
+	validateInputs=()=>{
+		const {inputArray} = this;
+		if(inputArray){
+			for (const key in inputArray) {
+				if (inputArray.hasOwnProperty(key) && inputArray[key].errorState)
```
[the remark this answers is not in the document]
Comment (vivekvijayakrishnan) [reply source/reactnative/src/adaptive-cards.js]: sure, we can remove this check completely.

File: source/reactnative/src/utils/enums.js (hunk 3 above)
Comment (vivekvijayakrishnan) [reply source/reactnative/src/utils/enums.js]: done

**Pull request merged**